libs/langchain/langchain/storage/upstash_redis.py

```diff
@@ -117,3 +118,57 @@ def yield_keys(self, *, prefix: Optional[str] = None) -> Iterator[str]:
                     yield relative_key
                 else:
                     yield key
+
+
+@deprecated("0.0.335", alternative="UpstashRedisByteStore")
```

Comment:
one nit: The new code seems to involve overhead encoding/decodings, feels a bit strange that the new alternative is less optimized from that point of view?

Reply:
Would you want to keep both then? IMO it's worth it to simplify/standardize how devs use it and I doubt anybody is at the moment since you have to write your own encoding and we don't document it